test/lint/lint-python.py

```diff
@@ -13,8 +13,10 @@
 import subprocess
 import sys
 
+# Customize mypy cache dir via environment variable
```

Comment:
This was added as part of #18210 but I didn't see an explanation given there. I think it is better to specify a location explicitly since the default is using the current directory which may be leading to unexpected results if this script is called from different contexts. But if we don't care about something like this, it may be removed.